packages/transitive-overlay/src/index.tsx

```diff
@@ -152,19 +155,36 @@ type MapImage = {
   options: { sdf?: boolean; content?: [number, number, number, number] };
 };
 
-const loadImages = (map: MapRef, images: MapImage[]) => {
-  images.forEach(img => {
-    map.loadImage(img.url, (error, image) => {
-      if (error) {
-        // eslint-disable-next-line no-console
-        console.error(`Error loading image ${img.id}:`, error);
-        return;
+const loadImages = async (map: MapRef, images: MapImage[]) => {
+  for (const img of images) {
+    // Only load if the image hasn't already been added
+    // @ts-ignore (TS may not know hasImage exists depending on the type you use)
+    if (!map.hasImage(img.id)) {
```

Comment:
@lightsage88 I have included/moved this condition in https://github.com/opentripplanner/otp-ui/pull/848/commits/a2bde30e41202083f6931721c6431dda37adb864. Let me know if that works for you.

Reply:
lgtm!